### DISCUSSION TITLE
Capacitor react app works fine on ipad but not on iphone

### DISCUSSION THREAD

**Discussion by mxye** [ionic-team/capacitor#5467] [Q&A]
So this issue has stumped me for days. I followed the [basic docs](https://capacitorjs.com/docs/ios) to make my react app into an ios app. Changed nothing, strictly followed the docs. The app runs on my physical ipad and all the ipad simulators provided by Xcode. All good there. However, on iphone and all the iphone simulators, it loads into a black screen. No error in the Xcode console. 

How is that even possible? I didn't specify "ipad" anywhere since it's not in the docs. The ios system as far as I know is also identical across the two types of devices. Just in case it's a screen size issue, I made the first page of the app a simple h1 tag, and it also only shows up on ipad/ipad simulators, but not on any of the iphone simulators. 

I read on stackoverflow ([this](https://stackoverflow.com/questions/34963740/ionic-app-wont-run-on-iphone-works-fine-on-ipad), and [this](https://stackoverflow.com/questions/4537460/iphone-app-works-fine-on-ipad-but-not-on-iphone)) it might be an Info.plist issue, but I never changed the auto-generated Info.plist. Plus the keys they mentioned on the two posts I found don't exist in my auto-generated Info.plist. 

I also checked my Capacitor versions (core, cli, ios) -- they are all the most recent one, 3.4.1. 

Anyone has any pointers?

**Comment by bfeelin** [comment]
I don't have physical devices to test on, but I am having the same issue. No error in the Xcode console, app loads into a white screen on iphone emulators. Works fine on Ipad.

Are you using [@codetrix-studio/capacitor-google-auth](https://www.npmjs.com/package/@codetrix-studio/capacitor-google-auth) also?

**Comment by bfeelin** [comment]
Check out this video on how to debug your iOS app using safari, it helped me solve my issue: https://www.youtube.com/watch?v=e6B5sH5J4cI

Once you have the debugger loaded, you can use window.location.reload() to restart the app and output any javascript errors.

`TypeError: undefined is not an object (evaluating 'gapi.iframes.getContext')` was the error I had. I'm using React & Firebase and recently added Capacitor. Googling that error lead me here: https://stackoverflow.com/questions/69778975/angular-ionic-mobile-app-ios-does-not-fetch-from-firebase-using-angularfire

I ended up adding the code below when initializing Firebase and it works on IPhones now. Not sure why it was only an issue on IPhones

```
if (Capacitor.isNativePlatform()) {
      initializeAuth(firebaseApp, {
        persistence: indexedDBLocalPersistence
      });
    }
```
